reference/forms/types/options/text_empty_data.rst.inc

```diff
@@ -0,0 +1,36 @@
+``empty_data``
+~~~~~~~~~~~~~~
+
+**type**: ``mixed``
+
+.. This file should only be included with start-after or end-before that's
+   set to this placeholder value. Its purpose is to let us include only
+   part of this file.
+
+DEFAULT_PLACEHOLDER
+
+This option determines what value the field will *return* when the submitted
+value is empty (or missing). It does not set an initial value if none is
+provided when the form is rendered in a view.
+
+This means it helps you handling form submission with blank fields. For
+example, if you want the ``name`` field to be explicitly set to ``John Doe``
+when no value is selected, you can do it like this::
+
+    $builder->add('name', null, [
+        'required'   => false,
+        'empty_data' => 'John Doe',
+    ]);
+
+This will still render an empty text box, but upon submission the ``John Doe``
+value will be set. Use the ``data`` or ``placeholder`` options to show this
```

Comment:
I wouldn't encourage the use of the inner `data` option for the default value. Cause that when it's a mapped field the `data` value set here will replace the bound value that comes from the underlying subject, thus it doesn't behave like a default value. There is already a warning note about it in 
https://symfony.com/doc/current/reference/forms/types/text.html#data

We should use the underlying subject to the root form to set default values or a `PRE_SET_DATA` event listener instead.

Reply:
and the `placeholder` option is only for display but it also doesn't actually set an initial value for this field

Reply:
I see, maybe that's something to fix from 4.4 (in another PR).